lib/context-pad/RefactoringsContextPadProvider.js

```diff
@@ -34,15 +36,22 @@ export default class RefactoringsContextPadProvider {
       }
 
       // pre-fetch results
-      this.refactorings.getRefactorings([ element ]);
+      this._refactorings.getRefactorings([ element ]);
 
       return {
         ...entries,
-        alert: {
-          action: () => {
-            const pad = this._contextPad.getPad(element);
+        refactorings: {
+          action: {
+            click: (event) => {
+              const pad = this._contextPad.getPad(element);
+              this._openPopupMenu(pad, element);
 
-            this._openPopupMenu(pad, element);
+              const entry = domClosest(event.delegateTarget || event.target, '.entry', true);
```

Comment:
🙈 Isn't this a hack? How does the replace popup menu to this? Active states should be a feature of the context pad.

Reply:
The method is consistent with how the new context pad handles active state:
https://github.com/camunda/improved-canvas/blob/7dc5052c7057f6ea1e5f7b08bd62a12e8a885e1e/lib/common/contextPad/ImprovedContextPadProvider.js#L110-L115

The "get the entry dom node from event" is consistent with the PopupMenu Code:
https://github.com/bpmn-io/diagram-js/blob/cc602c41a5b05510de4b81a753096919e47bc8b4/lib/features/popup-menu/PopupMenu.js#L585

Reply:
I don't think the context pad handles open state, as we never had this feature in the old context pad